wordpress-mobile/WordPress-iOS
# Media: MediaFileManager, refactored instance of an extension.

This PR brings in a refactor of what was previously `MediaLibrary+Local.swift`, an extension to the now defunct `MediaLibrary` (next PR).

Mostly moving code around and renaming, but particularly within the context of now being an actual class named `MediaFileManager`. This cleans things up quite nicely and clearly defines the dependency on choosing a directory type for any Media file operations. Ultimately, this nicely encapsulates media file and URL implementation into a single class.

Note:

- The `MediaDirectory` enum was moved outside of the scope of the class, previously the extension class.
- There are new class helper functions for continuing with the same pattern of "clearing the cache", as seen previously.

To test:
1. Make sure everything builds.
2. Make sure the tests pass.
3. Might require a clean 😅 

Needs review: @frosty, @SergioEstevao, mostly just refactoring for this one!
